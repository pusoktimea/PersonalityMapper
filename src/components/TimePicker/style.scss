@import '~compass-mixins/lib/compass';
@import "../../styles/_colors.scss";

$baseClass: '.activate-time-picker';

#{$baseClass} {
  .react-time-picker {
    &__button {
      @include border-radius(5px);
      background-color: transparent;
      border: 1px solid $mercury;
      height: 34px;
      &__input {
        padding: 6px 0px 6px 12px;
        
        input {
          color: $doveGray;
          font-size: 14px;
          line-height: 1.42857143;
          outline: none;
        }
      }
    }
    &__clear-button {
      display: none;
    }
    &__clock-button {
      @include border-radius(0 5px 5px 0);
      padding: 6px 12px;
      height: 34px;
      border: none;
      border-left: 1px solid $mercury;
      outline: none;
    }

    &__clock {
      display: none;
    }
  }

  &_icon {
    font-size: 14px;
    line-height: 1.42857143;
    color: $doveGray;
  }
}
