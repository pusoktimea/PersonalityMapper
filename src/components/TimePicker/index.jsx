import React, {Component} from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import TimePicker from 'react-time-picker';

import Icon from '../Icon';

import './style.scss';

class ActivateTimePicker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: props.value || new Date()
    };
  }

  onChange = (date) => {
    this.setState({
      value: date
    }, () => {
      if (typeof this.props.onChange === 'function') {
        this.props.onChange(date);
      }
    });
  }

  render() {
    const {value} = this.state;

    const {
      className
    } = this.props;

    const baseClass = 'activate-time-picker';
    return (
      <TimePicker
        className={cx(
          baseClass,
          className
        )}
        value={value}
        onChange={this.onChange}
        clearIcon={null}
        clockIcon={
          <Icon
            className={`${baseClass}_icon`}
            icon="clock-o"
          />
        }
      />
    );
  }
}

ActivateTimePicker.propTypes = {
  value: PropTypes.instanceOf(Date),
  onChange: PropTypes.func,
  className: PropTypes.string
};

export default ActivateTimePicker;
