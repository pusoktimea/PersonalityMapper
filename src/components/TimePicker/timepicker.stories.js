import React from 'react';
import {setAddon, storiesOf} from '@storybook/react';
import {withKnobs} from '@storybook/addon-knobs';
import JSXAddon from 'storybook-addon-jsx';

import TimePicker from './index';

setAddon(JSXAddon);
const stories = storiesOf('Components', module);

stories.addDecorator(withKnobs);
stories.addWithJSX('Time Picker', () => (
  <TimePicker />
));
