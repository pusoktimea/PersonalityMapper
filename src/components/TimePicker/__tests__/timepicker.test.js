import React from 'react';
import renderer from 'react-test-renderer';

import TimePicker from '../index';

describe('TimePicker test', () => {
  it('renders and matches snapshot', () => {
    expect(
      renderer.create(
        <TimePicker
          value={new Date('2017-12-25 00:00')}
        />
      ).toJSON()
    ).toMatchSnapshot();
  });
});
