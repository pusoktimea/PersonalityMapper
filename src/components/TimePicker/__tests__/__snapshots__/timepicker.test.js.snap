// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker test renders and matches snapshot 1`] = `
<div
  className="react-time-picker react-time-picker--closed activate-time-picker"
  onFocus={[Function]}
>
  <div
    className="react-time-picker__button"
  >
    <div
      className="react-time-picker__button__input"
    >
      <input
        max={null}
        min={null}
        name="time"
        onChange={[Function]}
        onFocus={[Function]}
        required={undefined}
        step={null}
        style={
          Object {
            "left": "-9999px",
            "position": "absolute",
            "top": "-9999px",
            "visibility": "hidden",
          }
        }
        type="time"
        value="00:00"
      />
      <input
        className="react-time-picker__button__input__hour"
        max={23}
        min={0}
        name="hour"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="--"
        required={null}
        type="number"
        value={0}
      />
      <span
        className="react-time-picker__button__input__divider"
      >
        :
      </span>
      <input
        className="react-time-picker__button__input__minute"
        max={59}
        min={0}
        name="minute"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="--"
        required={null}
        type="number"
        value={0}
      />
    </div>
    <button
      className="react-time-picker__clear-button react-time-picker__button__icon"
      onClick={[Function]}
      onFocus={[Function]}
      type="button"
    />
    <button
      className="react-time-picker__clock-button react-time-picker__button__icon"
      onBlur={undefined}
      onClick={[Function]}
      onFocus={[Function]}
      type="button"
    >
      <span
        className="activate-icon fa fa-clock-o activate-time-picker_icon"
        id={undefined}
        onClick={undefined}
      />
    </button>
  </div>
</div>
`;
